### PR TITLE
Event dispatcher

### PR DESCRIPTION
**Pull request by AyloSrd**

### CODE REVIEW

File: packages/event-dispatcher/src/eventDispatcher.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { create_custom_evt, get_event_listener_name } from "./utils"
+
+type Handlers<T> = {
+  [Property in keyof T as Property extends `on${infer EventName}` ? Uncapitalize<EventName> : never]: T[Property]
+}
+
+type DispatcherOptions = {
+  cancelable: boolean,
+}
```
Comment (thetarnav) [inline packages/event-dispatcher/src/eventDispatcher.ts:9]: Those types should probably be exported.

File: packages/event-dispatcher/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@solid-primitives/event-dispatcher",
+  "version": "0.0.100",
+  "description": "A template primitive example.",
+  "author": "Your Name <you@youremail.com>",
```
Comment (thetarnav) [inline packages/event-dispatcher/package.json:5]: Please add yourself as an author :)

File: packages/event-dispatcher/package.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "dist"
+  ],
+  "main": "./dist/server.cjs",
+  "module": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "exports": {
+    "node": {
+      "import": "./dist/server.js",
+      "require": "./dist/server.cjs"
+    },
```
Comment (thetarnav) [inline packages/event-dispatcher/package.json:37]: The exports mention a `server` export, but one isn't built.
It's important for the packages to work fine in SSR so using any client-only APIs (such as CustomEvent) should happen only in a client-only runtime.
To add one you only need to have a `server.ts` file in src with server fallback primitives. Tsup already has the correct setup.
See this as an example: https://github.com/solidjs-community/solid-primitives/blob/main/packages/devices/src/server.ts
Additionally, you can add a test for matching exports between `server` and `index`: https://github.com/solidjs-community/solid-primitives/blob/main/template/test/exports.test.ts

File: packages/event-dispatcher/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+</p>
+
+# @solid-primitives/event-dispatcher
+
+[![turborepo](https://img.shields.io/badge/built%20with-turborepo-cc00ff.svg?style=for-the-badge&logo=turborepo)](https://turborepo.org/)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/event-dispatcher?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/event-dispatcher)
+[![version](https://img.shields.io/npm/v/@solid-primitives/event-dispatcher?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/event-dispatcher)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
+
+`createEventoDispatcher` creates a custom-event dispatcher for emitting component events.
```
Comment (thetarnav) [inline packages/event-dispatcher/README.md:12]:
```suggestion
`createEventDispatcher` creates a custom-event dispatcher for emitting component events.
```

File: packages/event-dispatcher/README.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+npm install @solid-primitives/event-dispatcher
+# or
+yarn add @solid-primitives/event-dispatcher
+```
+
+## How to use it
+
+### Exemple: create and dispatch the event
+```tsx
+import { createEventDispatcher } from 'solid-create-event-dispatcher'
```
Comment (thetarnav) [inline packages/event-dispatcher/README.md:27]:
```suggestion
import { createEventDispatcher } from "@solid-primitives/event-dispatcher";
```

File: packages/event-dispatcher/package.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    "url": "git+https://github.com/solidjs-community/solid-primitives.git"
+  },
+  "bugs": {
+    "url": "https://github.com/solidjs-community/solid-primitives/issues"
+  },
+  "primitive": {
+    "name": "event-dispatcher",
+    "stage": 0,
+    "list": [
+      "createPrimitiveTemplate"
```
Comment (thetarnav) [inline packages/event-dispatcher/package.json:20]:
```suggestion
      "createEventDispatcher"
```

File: packages/event-dispatcher/package.json [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    ],
+    "category": "Display & Media"
```
Comment (thetarnav) [inline packages/event-dispatcher/package.json:22]:
```suggestion
    "category": "Utilities"
```

**Review by thetarnav — COMMENTED** [review]
The implementation looks good. And current functionality makes a good base to possibly extend further in the future.
All of the review comments touch only on other things that are necessary to publish the package.

**Comment by thetarnav** [comment]
Alright, this is great. Thank you for bringing this :)
Will publish the package now.

**Comment by AyloSrd** [comment]
Sorry for my late reply, thank you for the review!

**Pull request merged**